### DISCUSSION TITLE
Register component using only classname

### DISCUSSION THREAD

**Discussion by NoxxieNl** [livewire/livewire#1035] [4. General]
Hiya,

Currently I am creating an app where for the most of the time I am registering routes with `livewire::route()` method.

However the current syntax is that you need to alias the component name to let it find correctly. For IDE's and type hinting the correct class that is a nightmare.

So what I do now is to register the component like this:

```php
Livewire::component(Minvar::class, Minvar::class);
```
So that I can do this in my `web.php`:
```php
Route::livewire('minvar', Minvar::class);
```

Would be great If I can reduce the registering component to only one parameter:
```php
Livewire::component(Minvar::class);
```
and that the component get registerd with the full namespaced class name. I got the idea from the route changes to allow full namespacing classes ([Here](https://freek.dev/1210-a-better-way-to-register-routes-in-laravel)).

The only change what I can see that needs to be done to get this working is in `LivewireManager.php`:

```php
    public function component($alias, $viewClass = null)
    {
        $this->componentAliases[$alias] = $viewClass == null
                                          ? $alias
                                          : $viewClass;
    }
```

Happily to make a pull request to make this working. As what I can see using the way above it is not a breaking change, and the current setup using alias names will still work as expected :).

**Comment by mblarsen** [comment]
Seems like a useful enhancement to me.
